phpBB/phpbb/messenger/method/jabber.php

```diff
@@ -1,118 +1,259 @@
+	 *		->stream_options(
+	 *			'verify_peer' => true,
+	 *			'verify_peer_name' => true,
+	 *			'allow_self_signed' => false,
+	 *		);
+	 *
+	 * @return void
+	 */
+	public function init(): void
 	{
-		$this->connect_server		= ($server) ? $server : 'localhost';
-		$this->port					= ($port) ? $port : 5222;
+		$this->username($this->config['jab_username'])
+			->password($this->config['jab_password'])
+			->ssl((bool) $this->config['jab_use_ssl'])
+			->server($this->config['jab_host'])
+			->port($this->config['jab_port'])
+			->stream_options['ssl'] = [
+				'verify_peer' => $this->config['jab_verify_peer'],
+				'verify_peer_name' => $this->config['jab_verify_peer_name'],
+				'allow_self_signed' => $this->config['jab_allow_self_signed'],
+			];
+	}
 
-		// Get the server and the username
-		if (strpos($username, '@') === false)
+	/**
+	 * {@inheritDoc}
+	 */
+	public function get_id(): int
+	{
+		return NOTIFY_IM;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function get_queue_object_name(): string
+	{
+		return 'jabber';
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function is_enabled(): bool
+	{
+		return
+			!empty($this->config['jab_enable']) &&
+			!empty($this->config['jab_host']) &&
+			!empty($this->config['jab_username']) &&
+			!empty($this->config['jab_password']);
+	}
+
+	/**
+	 * Set ssl context options
+	 * See http://php.net/manual/en/context.ssl.php
+	 *
+	 * @param array $options SSL context options array
+	 * @return self
+	 */
+	public function stream_options(array $options = []): self
+	{
+		if ($this->use_ssl)
 		{
-			$this->server = $this->connect_server;
-			$this->username = $username;
+			// Change default stream options if needed
+			$this->stream_options['ssl'] = array_merge($this->stream_options['ssl'], $options);
 		}
-		else
-		{
-			$jid = explode('@', $username, 2);
 
-			$this->username = $jid[0];
-			$this->server = $jid[1];
-		}
+		return $this;
+	}
 
-		$this->password				= $password;
-		$this->use_ssl				= ($use_ssl && self::can_use_ssl()) ? true : false;
-		$this->verify_peer			= $verify_peer;
-		$this->verify_peer_name		= $verify_peer_name;
-		$this->allow_self_signed	= $allow_self_signed;
+	/**
+	 * Set password to connect to server
+	 *
+	 * @param string $password Password to connect to server
+	 * @return self
+	 */
+	public function password(string $password = ''): self
+	{
+		$this->password	= html_entity_decode($password, ENT_COMPAT);
+
+		return $this;
+	}
+
+	/**
+	 * Set use of ssl to connect to server
+	 *
+	 * @param bool $use_ssl Flag indicating use of ssl to connect to server
+	 * @return self
+	 */
+	public function ssl(bool $use_ssl = false): self
+	{
+		$this->use_ssl = $use_ssl && self::can_use_ssl();
+
+		return $this;
+	}
+
+	/**
+	 * Set port to connect to server
+	 * use_ssl flag should be set first
+	 *
+	 * @param int $port Port to connect to server
+	 * @return self
+	 */
+	public function port(int $port = 5222): self
+	{
+		$this->port	= ($port) ? $port : 5222;
```

Comment:
I guess checking > 0 would make more sense than checking whether it evaluates to true.